cyipopt/tests/unit/test_scipy_optional.py

```diff
@@ -131,6 +135,94 @@ def test_minimize_ipopt_jac_hessians_constraints_with_arg_kwargs():
     np.testing.assert_allclose(res.get("x"), expected_res, rtol=1e-5)
 
 
+@pytest.mark.skipif("scipy" not in sys.modules,
+                    reason="Test only valid if Scipy available.")
+@pytest.mark.parametrize('method', MINIMIZE_METHODS)
+def test_minimize_ipopt_jac_with_scipy_methods(method):
+    x0 = [0] * 4
+    a0, b0, c0, d0 = 1, 2, 3, 4
+
+    def fun(x, a=0, e=0, b=0):
+        assert a == a0
+        assert b == b0
+        fun.count += 1
+        return (x[0] - a) ** 2 + (x[1] - b) ** 2 + x[2] ** 2 + x[3] ** 2
+
+    def grad(x, a=0, e=0, b=0):
+        assert a == a0
+        assert b == b0
+        grad.count += 1
+        return [2 * (x[0] - a), 2 * (x[1] - b), 2 * x[2], 2 * x[3]]
+
+    def hess(x, a=0, e=0, b=0):
+        assert a == a0
+        assert b == b0
+        hess.count += 1
+        return 2 * np.eye(4)
+
+    def fun_constraint(x, c=0, e=0, d=0):
+        assert c == c0
+        assert d == d0
+        fun_constraint.count += 1
+        return [(x[2] - c) ** 2, (x[3] - d) ** 2]
+
+    def grad_constraint(x, c=0, e=0, d=0):
+        assert c == c0
+        assert d == d0
+        grad_constraint.count += 1
+        return np.hstack((np.zeros((2, 2)),
+                         np.diag([2 * (x[2] - c), 2 * (x[3] - d)])))
+
+    fun.count = 0
+    grad.count = 0
+    hess.count = 0
+    fun_constraint.count = 0
+    grad_constraint.count = 0
+
+    constr = {
+        "type": "eq",
+        "fun": fun_constraint,
+        "jac": grad_constraint,
+        "args": (c0,),
+        "kwargs": {'d': d0},
+    }
+
+    kwargs = {}
+    jac_methods = {'cg', 'bfgs', 'newton-cg', 'l-bfgs-b', 'tnc', 'slsqp,',
+                   'dogleg', 'trust-ncg', 'trust-krylov', 'trust-exact',
+                   'trust-constr'}
+    hess_methods = {'newton-cg', 'dogleg', 'trust-ncg', 'trust-krylov',
+                   'trust-exact', 'trust-constr'}
+    constr_methods = {'slsqp', 'trust-constr'}
```

Comment:
Theoretically, `COBYLA` accepts inequality constraints, but there are issues with SLSQP convergence when I convert the constraints above to inequality constraints, and COBYLA doesn't pass the test anyway. Just keep it simple. In a sense, only one method is really _required_ to check that the `minimize_ipopt` code is doing the right thing (e.g. callables are wrapped properly)... the rest just make me feel more comfortable.